src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -1223,12 +1224,22 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             },
             Err(_) => "move |<args>| <body>".to_string()
         };
-
+        let kind = if let Some(generator_kind) = use_span.generator_kind() {
+            match generator_kind {
+                GeneratorKind::Async(_) => "async block",
```

Comment:
This should match on `AsyncGeneratorKind` (the field inside `Async(_)`) to differentiate async blocks, closures, and functions (though I'd expect only the former two to appear here).